lib/ansible/plugins/filter/core.py

```diff
@@ -342,6 +343,37 @@ def combine(*terms, **kwargs):
     return result
 
 
+SEMVER_VALID_KEYS = ['major', 'minor', 'patch', 'prerelease', 'buildmetadata']
+# This list can be generated from the SEMVER_RE expression by
+# `SEMVER_VALID_KEYS=list(SEMVER_RE.groupindex.keys())` but is expected to stay static
+
+
+def parse_semver(
+        vstring,
+        keys=None
```

Comment:
What is the reason for adding keys here? The keys needed could be chosen after using the filter.

Reply:
My idea was to support directly asking for a part like the url filter enables